### PR TITLE
Consolidates solution/projects and moves to Azure DevOps

### PR DESCRIPTION
**Pull request by danielwertheim**
Main changes are:
  
  - Using .NET Core SDK style projects to multitarget
    .NET4.5 and .NET Standard 1.6
  - Organizes folders and projects
  - Adds Directory.builds.props to unify props in all projects
  - Splits tests into IntegrationTests and UnitTests
  - Moves signing key to root
  - Fixes pointing to correct logo at CNCF artwork in NuGet
  - Uses Azure Pipeline (YAML) for CI
  - Starts using TestCollection to control paralellization
  - Uses Global.json to lock to certain SDK
    (be as similar as possible at build server)

### CODE REVIEW

**Comment by danielwertheim** [comment]
@ColinSullivan1 Might need to update the pipeline after this has reached `master` as it points to the `azure-pipeline.yml` file in the branch for this PR.

As of now, all none `master` builds will generate a pre-release build (artifact can be downloaded in the build in the pipeline). Hence we currently does not publish it anywhere. There's a disabled step in the YAML file fir this.

The Version in itself is currently controlled by bumping in the YAML file.

We can look into improvements later on in smaller PRs and/or evaluate GitHub Actions.

**Review by watfordgnf — APPROVED** [review]
No concerns from me, looks great.

**Comment by ColinSullivan1** [comment]
I've confirmed the command line build, am setting up VS 2019 to confirm the project - will get that done today.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  - task: DownloadGitHubRelease@0
+    displayName: 'Download latest Win64 NATS-Server'
+    inputs:
+      connection: 'NATS-CI'
+      userRepository: 'nats-io/nats-server'
+      itemPattern: 'nats-server-**-windows-amd64.zip'
+      downloadPath: 'nats-server'
+  
+  - task: ExtractFiles@1
+    displayName: 'Extract NAST-Server files '
```
Comment (ColinSullivan1) [inline azure-pipelines.yml]: NATS...

File: README.md [hunk 2]
```diff
@@ -1,90 +1,69 @@
...
+The repository contains several projects, all located under `src\`
 
-For both .NET 4.5 and .NET core, there are simple batch files.  These will build the respective NATS.Client.dll and the provided examples with only requiring the  .NET framework SDK/Platform
-
-#### .NET 4.5
-
-Ensure you have installed the .NET Framework 4.5.1 or greater.
+* NATS - The NATS.Client assembly
+* Tests
+    * IntegrationTests - XUnit tests, verifying the client integration with `nats-server.exe` (ensure you have `nast-server.exe` in your path to run these).
```
Comment (ColinSullivan1) [inline README.md]: nats-server.exe

**Comment by danielwertheim** [comment]
@ColinSullivan1 I do believe you should be able to use VS2017 if you have that installed.

File: README.md (hunk 2 above)
Comment (danielwertheim) [reply README.md]: Fixed

File: azure-pipelines.yml (hunk 1 above)
Comment (danielwertheim) [reply azure-pipelines.yml]: Fixed

File: src/Tests/IntegrationTests/TestConnection.cs [hunk 3]
```diff
@@ -1,1103 +1,1104 @@
...
+                    args.SignedNonce = kp.Sign(args.ServerNonce);
+                };
+                Options opts = ConnectionFactory.GetDefaultOptions();
+                opts.SetUserCredentialHandlers(jwtEh, sigEh);
+                new ConnectionFactory().CreateConnection(opts).Close();
+            }
+        }
+
+        [Fact]
+        public void TestJWTFunctionality()
```
Comment (danielwertheim) [inline src/Tests/IntegrationTests/TestConnection.cs:357]: @ColinSullivan1 Can we remove empty tests like these or should the be "Skipped" with e.g. "WorkInProgress"?

Comment (ColinSullivan1) [reply src/Tests/IntegrationTests/TestConnection.cs:357]: These can be skipped - WIP.  Thanks - I missed that one.

**Comment by ColinSullivan1** [comment]
LGTM.  Thanks!  Building with all green tests on my VM.

**Pull request merged**